lib/server/controllers/qcFlagTypes.controller.js

```diff
@@ -37,6 +40,43 @@ const getQcFlagTypeByIdHandler = async (req, res) => {
     }
 };
 
+// eslint-disable-next-line valid-jsdoc
+/**
+ * List All QCFlagReasons
+ */
+const listQCFlagTypesHandler = async (req, res) => {
+    const validatedDTO = await dtoValidator(
+        DtoFactory.queryOnly({
+            filter: {
+                ids: Joi.array().items(Joi.number()),
+                bad: Joi.boolean(),
+                archived: Joi.boolean(),
+                names: Joi.alternatives().try(Joi.array().items(Joi.string()), Joi.object({ like: Joi.array().items(Joi.string()) })),
+                methods: Joi.alternatives().try(Joi.array().items(Joi.string()), Joi.object({ like: Joi.array().items(Joi.string()) })),
```

Comment:
What we defined for run numbers in runs overview is:
- if we have only one token, use it as substring
- if we have a comma separated list of items, we match them exactly

It may probably be improved, but I suppose we should try to be uniform?

Reply:
applied